heareval/predictions/task_predictions.py

```diff
@@ -253,6 +254,40 @@ def validation_step(self, batch, batch_idx):
     def test_step(self, batch, batch_idx):
         return self._step(batch, batch_idx)
 
+    def log_scores(self, name: str, score_args):
+        """Logs the metric score value for each score defined for the model"""
+        assert hasattr(self, "scores"), "Scores for the model should be defined"
+        end_scores = {}
+        for score in self.scores:
+            score_ret = score(*score_args)
+            validate_score_return_type(score_ret)
+            # If the returned score is a tuple, store each subscore as separate entry
+            if isinstance(score_ret, tuple):
+                # The first score in the returned tuple will be used
+                # as primary score for this metric
+                end_scores[f"{name}_{score}"] = score_ret[0][1]
```

Comment:
Not sure if this is the right PR to change this, but I hate this JSON schema I defined.

It looks like this:

```
    "test": {
        "test_score": 0.5070093274116516,
        "test_loss": 11.01280689239502,
        "test_event_onset_200ms_fms": 0.5070093274116516,
        "test_segment_1s_er": 0.5021833777427673,
        "validation_score": 0.4511459767818451,
        "epoch": 470,
        "time_in_min": 64.55239222049713
    },
```

When I want to crawl over this JSON and find ALL the scores, I resort to the hacky trick of seeing if the string starts with "test" and isn't "test_score" or "test_loss". Kinda gross.

I would prefer ["test"]["test_scores"] = {"onset_fms": ..., ...} etc in addition to ["test"]["test_score"] and ["test"]["test_loss"].

Maybe we should make an issue for this?

Reply:
#336